test/index.js

```diff
@@ -54,192 +55,244 @@ describe("getEnvironmentVariables", () => {
+			};
 
-		const expectedVariables = {
-			"VAR1": "1",
-			"VAR2": "2",
-			"VAR4": "1=2=3=4",
-			"VAR5": "   1 2   3 ",
-			"VAR6": "1 2 3"
-		};
+			const execSyncCommands = [];
+			childProcess.execSync = (command) => {
+				execSyncCommands.push(command);
+				if (command.indexOf("node") !== -1) {
+					throw new Error("Unable to find node");
+				}
 
-		let childProcess = require("child_process");
-		childProcess.execSync = (command) => {
-			return _.map(expectedVariables, (value, key) => {
-				return `${key}=${value}`;
-			}).join('\n');
-		};
+				return _.map(defaultExpectedVariables, (value, key) => {
+					return `${key}=${value}`;
+				}).join('\n');
+			};
 
-		const actualResult = index.getEnvironmentVariables();
-		assert.deepEqual(actualResult, expectedVariables);
-	});
+			let loggedWarnings = [];
 
-	it("prints warning when environment variable does not match expected format", () => {
-		fs.existsSync = (filePath) => {
-			return filePath !== etcPaths;
-		};
+			const originalConsoleLog = console.log;
+			console.log = (data) => {
+				loggedWarnings.push(data);
+			};
 
-		const expectedVariables = {
-			"": "1"
-		};
+			const actualResult = index.getEnvironmentVariables();
 
-		childProcess.execSync = (command) => {
-			return _.map(expectedVariables, (value, key) => {
-				return `${key}=${value}`;
-			}).join('\n');
-		};
+			console.log = originalConsoleLog;
 
-		let loggedWarnings = [];
+			assert.deepEqual(loggedWarnings.length, 1);
+			assert.isTrue(loggedWarnings[0].indexOf("node") !== -1);
+			assert.deepEqual(actualResult, defaultExpectedVariables);
+		});
+
+		it("gets the process env written in a file by executed node process", () => {
+			fs.existsSync = (filePath) => {
+				return filePath !== etcPaths;
+			};
 
-		const originalConsoleLog = console.log;
-		console.log = (data) => {
-			loggedWarnings.push(data);
-		};
 
-		const actualResult = index.getEnvironmentVariables();
+			const execSyncCommands = [];
+			childProcess.execSync = (command) => {
+				execSyncCommands.push(command);
+				if (command.indexOf("node") !== -1) {
+					return "";
+				}
 
-		console.log = originalConsoleLog;
+				throw new Error("You shouldn't get here!");
+			};
 
-		assert.deepEqual(loggedWarnings.length, 1);
+			fs.readFileSync = (filePath, encoding) => JSON.stringify(defaultExpectedVariables);
 
-		assert.isTrue(loggedWarnings[0].indexOf("does not match") !== -1);
+			const actualResult = index.getEnvironmentVariables();
+			assert.deepEqual(actualResult, defaultExpectedVariables);
+		});
 	});
 
-	it("returns process.env on windows", () => {
-		fs.existsSync = (filePath) => true;
+	describe("when env is used", () => {
+		it("returns correct variables", () => {
+			fs.existsSync = (filePath) => {
+				return filePath !== etcPaths;
+			};
 
-		const expectedVariables = {
-			"VAR1": "1"
-		};
+			childProcess.execSync = (command) => {
+				return _.map(defaultExpectedVariables, (value, key) => {
```

Comment:
You can shorten the whole thing here but I think I'd prefer `_.map`'s second argument shortened only